.github/workflows/release.yml

```diff
@@ -0,0 +1,67 @@
+name: Release
+on:
+  push:
+    branches:
+      - master
+permissions: write-all
+jobs:
+  build:
+    name: Release
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup rust toolchain
+        run: rustup toolchain install stable --profile minimal
+
+      - name: Setup rust cache
+        uses: Swatinem/rust-cache@v2
+
+      - name: Get version
+        id: get_version
+        run: echo "version=$(cargo read-manifest | jq -r '.version')" >> $GITHUB_OUTPUT
+
+      - name: Check if release exists
```

Comment:
why not break this up into an initial job that makes the build & release one conditional?

https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs